src/app/action-items/action-items.component.ts

```diff
@@ -0,0 +1,52 @@
+import { TasksService } from './../tasks.service';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+
+@Component({
+  selector: 'action-items',
+  templateUrl: './action-items.component.html',
+  styleUrls: ['./action-items.component.css']
+})
+export class ActionItemsComponent implements OnInit, OnDestroy {
+  color = '';
+  loading = true;
+  displayedColumns: string[] = ['title', 'type', 'completed', 'dueDate', 'daysLeft'];
+  dataSource;
+  dataSourceFetchedAgain;
+  private subscription: Subscription;
+
+  constructor(private tasksService: TasksService) {
+    this.subscription = this.tasksService.getAllTasks().subscribe(tasks => {
+      this.dataSource = tasks;
+      this.loading = false;
+    });
+    this.dataSourceFetchedAgain = this.tasksService.passTableData();
+  }
+
+  ngOnInit() {
+    this.tasksService.getCurrentDate();
+    // here console log prints undefined, that's why I fetched tableData from tasksService again
```

Comment:
It is because tasks are loaded asynchronously and they are available after 1 second.